src/main/java/net/neoforged/neoforge/common/util/NbtListCollector.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.common.util;
+
+import com.google.common.collect.ImmutableSet;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.function.BinaryOperator;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collector;
+import net.minecraft.nbt.ListTag;
+import net.minecraft.nbt.Tag;
+
+/**
+ * Stream support for converting a stream of NBT tags into an NBT ListTag.
+ * Useful for stream operations and mapping functions.
+ *
+ * Usage: {@code Stream.doStuff().collect(NbtListCollector.toNbtList())}
+ */
+public class NbtListCollector implements Collector<Tag, ListTag, ListTag> {
```

Comment:
Should this be deprecated as well, or is it intended to be used after the legacy stuff is removed?

Reply:
This is a utility I used a TON with saved data and NBT code; it collects a stream of NBT tags into a `ListTag`. As long as NBT operations exist, I figure people can use this.